### PR TITLE
kvm: fixed hardcoded parameters of cpus and mem

### PR DESCRIPTION
**Pull request by ppalucki**
Signed-off-by: Daniel Tadrzak daniel.tadrzak@intel.com

### CODE REVIEW

File: stage1/init/init.go [hunk 1]
```diff
@@ -48,6 +49,16 @@ package main
 //   return sd_pid_get_slice(pid, slice);
 // }
 //
+// long
+// getTotalMemory()
```
Comment (ppalucki) [inline stage1/init/init.go]: please be concise with existing C function naming s/getTotalMemoery/get_total_memory/ 
and add quick docstring (I even don't now is it returns b/kb/mb)

File: stage1/init/init.go [hunk 2]
```diff
@@ -263,9 +274,23 @@ func getArgsEnv(p *Pod, flavor string, debug bool, n *networking.Networking) ([]
 			return nil, nil, err
 		}
 
-		// TODO: base on resource isolators
-		cpu := 1
-		mem := 128
+		manifest := p.Manifest
```
Comment (ppalucki) [inline stage1/init/init.go]: why creating new variable here ?

File: stage1/init/kvm/manifest.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+// limitations under the License.
+
+package kvm
+
+import (
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+)
+
+func GetMemAndCpusFromIsolators(isolators types.Isolators) (int64, int64) {
```
Comment (ppalucki) [inline stage1/init/kvm/manifest.go]: docstring missing

File: stage1/init/kvm/manifest.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		case *types.ResourceCPU:
+			cpusQuantity := v.Limit()
+			cpus += cpusQuantity.Value()
+		}
+	}
+
+	return mem, cpus
+}
+
+func GetMemAndCpusFromManifest(podManifest *schema.PodManifest) (int64, int64) {
```
Comment (ppalucki) [inline stage1/init/kvm/manifest.go]: why from manifest not from apps directly ? 

[please read](https://en.wikipedia.org/wiki/Law_of_Demeter) or [this](http://c2.com/cgi/wiki?LawOfDemeter)

File: stage1/init/kvm/manifest.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	total_cpus := int64(0)
+	total_mem := int64(0)
```
Comment (ppalucki) [inline stage1/init/kvm/manifest.go]: replace with

```
var (
  total_cpus int64
  total_mem int64
)
```

File: stage1/init/init.go [hunk 6]
```diff
@@ -263,9 +274,23 @@ func getArgsEnv(p *Pod, flavor string, debug bool, n *networking.Networking) ([]
...
+
+		// if user doesn't specify amount of cpus we set no limit.
+		if cpu == 0 {
+			cpu = int64(runtime.NumCPU())
+		}
+		// Convert bytes into megabytes
+		mem /= 1024 * 1024
+
+		// If user doesn't specify amount of mem we set 1GB.
+		// else add additional 128MB for system process.
```
Comment (ppalucki) [inline stage1/init/init.go]: please add note in docs about this to `Documentation/running-lkvm-stage1.md`

**Comment by jellonek** [comment]
LGTM

File: Documentation/running-lkvm-stage1.md [hunk 7]
```diff
@@ -77,6 +77,8 @@ If you want to see the kernel and boot messages, run rkt with the `--debug` flag
 
 You can exit pressing `<Ctrl-a x>`.
 
+Notes: By default containers start working on all cpus and with 1GB. If user specify parameters via pod-manifest, memory will be increased by 128MB. Additional memory is required by system process.
```
Comment (ppalucki) [inline Documentation/running-lkvm-stage1.md]: parameters/resource isolators

processes (e.g. kernel)

File: stage1/init/kvm/resources.go [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+// limitations under the License.
+
+package kvm
+
+import (
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+)
+
+// findResources find value of last isolator for particular type.
```
Comment (ppalucki) [inline stage1/init/kvm/resources.go]: s/find/finds

File: stage1/init/kvm/resources.go [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+		case *types.ResourceCPU:
+			cpusQuantity := v.Limit()
+			cpus = cpusQuantity.Value()
+		}
+	}
+
+	return mem, cpus
+}
+
+// GetAppsResources return values specfied by user in pod-manifest.
```
Comment (ppalucki) [inline stage1/init/kvm/resources.go]: s/return/returns

File: stage1/init/init.go [hunk 10]
```diff
@@ -263,9 +262,22 @@ func getArgsEnv(p *Pod, flavor string, debug bool, n *networking.Networking) ([]
...
+		if mem == 0 {
+			mem = int64(1024)
```
Comment (ppalucki) [inline stage1/init/init.go]: 1024

File: Documentation/running-lkvm-stage1.md (hunk 7 above)
Comment (jellonek) [inline Documentation/running-lkvm-stage1.md]: System process? Which?
"by system" is enough.

If user specifies resource isolators - memory will be increased by 128MB above summary value of particular applications.

What if one (for example from ten) application has specified memory isolator, and rest of them does not have such restrictions? Whole pod will be limited to these application limit + 128MB for system/rest? It looks quite bad.

**Pull request closed without merging**